CMake/TestForSTB.cmake

```diff
@@ -0,0 +1,37 @@
+# Author: Omar Shrit
+
+#[=======================================================================[.rst:
+TestForSTB
+--------------
+
+Test to verify if the last version of STB that contains static
+functions is available in the system
+
+check if the compiler supports the standard ANSI sstream header
+
+::
+
+  CMAKE_HAS_STATIC_STB - defined by the results
+#]=======================================================================]
+
+if(NOT DEFINED CMAKE_HAS_STATIC_STB)
+  message(CHECK_START "Check for stb")
+  try_compile(CMAKE_HAS_STATIC_STB stb/main.cpp
```

Comment:
Does this need to be `try_compile(CMAKE_HAS_STATIC_STB SOURCES stb/main.cpp stb/alib.cpp stb/blib.cpp ...)`?  I think otherwise this looks right!  I haven't tested it locally yet though.